src/app-layer-htp.c

```diff
@@ -689,6 +689,11 @@ static void HTPHandleError(HtpState *s, const uint8_t dir)
 
     size_t size = htp_list_size(s->conn->messages);
     size_t msg;
+    if(size >= 0x10000) {
+        DEBUG_VALIDATE_BUG_ON("Overflowing htp_messages_offset");
```

Comment:
is this correct use of `assert`?

Also please avoid using hex notation of numbers to improve readability. In this case the value should probably be `USHRT_MAX`, to indicate that its about the max value the type can hold.

Reply:
It also seems likely that this limit is way to high. IIRC the events we read from htp are in an array. This would be a very big array. We should probably bail much earlier, with a special "final" event indicating that we've reached the per flow event limit, and we will produce no further events.

Reply:
> is this correct use of assert?

I have seen it before and it works (as a pointer to a string is not 0) but I do not know if the style is a good habit

> USHRT_MAX

Ok for that

> It also seems likely that this limit is way to high.

What would be a better limit ? 1024 ?

Reply:
Something like that. Maybe even less, like 512 or even 256.

Reply:
Going for 512.
We'll see if fuzzing finds a way to get them